### DISCUSSION TITLE
Interaural time delays for a much more natural stereophonic experience helping you to hear out all musicians

### DISCUSSION THREAD

**Discussion by LucasMens** [jamulussoftware/jamulus#1553] [Ideas]
Hi Volker, and all developers, my great respect and many many thanks for creating Jamulus which has provided so many hours of delight while singing with my choir!

I want to suggest an addition to the signal processing to improve the stereo experience for Jamulus users. If this has been discussed before, sorry.

Our choir now uses stereo in and out and a specific Balance/pan settings for the soprano-, alto-, tenor- and bass voices which makes it a lot easier to discern all voices (instruments) compared to mono. However, it does not sound very natural. The thing is that now, only **level** differences (Interaural Level Differences, ILD) are used to create a stereo effect. Full panning now sets one channel output to zero which sounds a bit awkward, and also has the disadvantage of making the panned voices some 6 dB softer than voices that are allowed to remain in the middle. This can be improved a lot by adding **timing** differences, or Interaural Time Differences (ITD). This will increase the perceived spatial separation of voices/instruments even more than ILDs do. 

ILD and ITD is discussed in https://en.wikipedia.org/wiki/Sound_localization#ITD_and_IID). At https://auditoryneuroscience.com/spatial-hearing/binaural-cues you can listen to the powerful effect of ITDs, even without any ILD. 

ITDs are 0 for sounds in front of the listeners and + or – 625 us for the full L (-90 deg) and R (+90 deg) position. Ideally, changing the balance/pan would affect the time difference and the level difference. Due to the geometry of the human head and physics,the relation between azimuth and ILD is a bit complex and frequency dependent (https://www.researchgate.net/figure/ILDs-as-a-function-of-frequency-for-six-different-azimuths-Circles-show-KEMAR_fig2_283110251), but I guess a simplified relation would do: 15 dB attenuation at full pan. An option to disable ILD and use only ITD to prevent panned voices becoming too soft would be nice.

Of course, introducing ITD implies extra signal processing which may or may not disrupt streaming, I have no idea. 

It is amazing that Jamulus is completely open source and free. I would gladly pay for a slick commercial version. Heck, I would promote such a version for on-line discussions. Video conferencing has its merits, but the accurate synchronous audio experience in Jamulus, especially with spatial separation in stereo, to me seems something that would fill a pretty big niche.

Best regards,

Lucas Mens

**Comment by dcorson-ticino-com** [comment]
Hi Lucas,
You will not have to wait long for this.  
It should be coming in the next version, if all goes to plan.
D
